### PR TITLE
[fix] google engine: results XPath

### PR DESCRIPTION
**Pull request by tiekoetter**
## What does this PR do?

Fix the google engine results XPath.

## Why is this change important?

Google would not working anymore.

## How to test this PR locally?

1. `make run`
2. Search `!go time`

### CODE REVIEW

**Comment by return42** [comment]
First: Since it does not have a effect, I reverted 747cf1a24 (in c72d70d45) and pushed it on the branch of this PR.  @tiekoetter @unixfox I hope this is OK for you.

**I think we have two different problems with google ..**

- one is CONSENT cookie we track in https://github.com/searxng/searxng/issues/1555
- the other is the XPath problem (what XPath fits to a google response?).

This PR is about the XPath problem and **this PR is a good interim solution but I think we will need to put more work on the XPath**.

**About XPath:**

My experience is, that google has two different HTML results, there is the "first page" and the "following pages"

- The "first page" includes (more) advertising and various layouts for grouping topics (e.g. "Headlines", "maps" and more). These layouts have two effects:
  - topics mess the list of results we are looking for and make it hard to find the "real results" we are looking for
  - since the first page is full of topics (and advertising) there are less results, sometimes I see only three or four "real results" on google's first page and all the other links are related to topics and advertising. 

- The "following pages" are more like a *"real result list"* that contains less advertising and topics

This becomes clear if you search for *comercially interesting terms* on google, by example:

In SearxNG search for `!go smartphone` / in my language (DE) SearXNG build this google request:

- https://www.google.de/search?q=smartphone&hl=de&lr=lang_de&ie=utf8&oe=utf8&start=0&filter=0

.. be warned the result from google also depends on the region code your browser places in the HTTP header (my FFox sends `Accept-Language: de,en;q=0.5`).

If you open the link on google and compare the "first page" with the "following pages" you see what I described above.

**To summarize:**

1. This PR is huge step forward to improve the google results.
2. We need to invest more time in the **CONSENT cookie** and the **XPath problem**

I'm going to merge this PR ..
